src/plugins/orangehrmDashboardPlugin/Api/EmployeeOnLeaveAPI.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Dashboard\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Dashboard\Api\Model\EmployeeOnLeaveModal;
+use OrangeHRM\Dashboard\Traits\Service\EmployeeOnLeaveServiceTrait;
+use OrangeHRM\Leave\Dto\LeaveListSearchFilterParams;
+
+class EmployeeOnLeaveAPI extends Endpoint implements CollectionEndpoint
+{
+    use EmployeeOnLeaveServiceTrait;
+
+    public const DATE = 'date';
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $leaveListSearchFilterParams = new LeaveListSearchFilterParams();
```

Comment:
```suggestion
        $employeeOnLeaveSearchFilterParams = new EmployeeOnLeaveSearchFilterParams();
```
:thinking:  can refactor rename